docs/guide/declarative_containers.md

```diff
@@ -0,0 +1,111 @@
+# Declarative Environment Configuration
+
+jobq allows you to define your Docker environment in a declarative YAML format. Instead of writing a Dockerfile directly.
+
+## YAML Structure
+
+The YAML has the following structure.
```

Comment:
Leading with an example is a good idea! I would then structure the rest of the page by the individual fields that the user can set (so we get a usable page sidenav) and mention which ones are mandatory.


Additionally, I would add a link to the official Dockerfile reference (https://docs.docker.com/reference/dockerfile/), to aid in understanding the meaning of the different options.

Since we currently mirror the structure of a "regular" Dockerfile, with abstractions only in the form of the `dependencies` map, understanding Dockerfiles is still somewhat required knowledge.

Reply:
Agree.

Though, if we only have that minor of an abstraction around dockerfiles themselves and we still require underlying docker knowledge, we could think about scrapping `.yaml` support entirely (or extending it such that it actually hides complexity). 

Because, as of now, i don't see that much upside in the `.yaml` themselves. Dockerfiles are also version controlled. 
Or am I missing something?

Reply:
Yeah, that is kind of what I wanted to probe with this PR: see how much (useful) abstraction over Dockerfiles we currently provide.

There is some (dependency management), but a lot of the underlying piping is still shining through.

These insights can guide whether we want to elaborate on the image building at some point or ditch it in favor of the user having to write their own Dockerfiles again.

(I really like the idea of the feature, but at the moment it really is stuck in an uncanny valley of sorts)